python/dgl/nn/pytorch/explain/pgexplainer.py

```diff
@@ -358,3 +356,227 @@ def explain_graph(self, graph, feat, tmp=1.0, training=False, **kwargs):
         self.clear_masks()
 
         return (probs, edge_mask) if training else (probs.data, edge_mask)
+
+
+class HeteroPGExplainer(PGExplainer):
+    r"""PGExplainer from `Parameterized Explainer for Graph Neural Network
+    <https://arxiv.org/pdf/2011.04573>`__, adapted for heterogeneous graphs
+
+    PGExplainer adopts a deep neural network (explanation network) to parameterize the generation
+    process of explanations, which enables it to explain multiple instances
+    collectively. PGExplainer models the underlying structure as edge
+    distributions, from which the explanatory graph is sampled.
+
+    Parameters
+    ----------
+    model : nn.Module
+        The GNN model to explain that tackles multiclass graph classification
+
+        * Its forward function must have the form
+          :attr:`forward(self, graph, nfeat, embed, edge_weight)`.
+        * The output of its forward function is the logits if embed=False else
+          the intermediate node embeddings.
+    num_features : int
+        Node embedding size used by :attr:`model`.
+    coff_budget : float, optional
+        Size regularization to constrain the explanation size. Default: 0.01.
+    coff_connect : float, optional
+        Entropy regularization to constrain the connectivity of explanation. Default: 5e-4.
+    sample_bias : float, optional
+        Some members of a population are systematically more likely to be selected
+        in a sample than others. Default: 0.0.
+    """
+
+    def train_step(self, graph, feat, tmp, **kwargs):
+        r"""Compute the loss of the explanation network
+
+        Parameters
+        ----------
+        graph : DGLGraph
+            Input batched heterogeneous graph.
+        feat : dict[str, Tensor]
+            A dict mapping node types (keys) to feature tensors (values).
+            The input features are of shape :math:`(N_t, D_t)`. :math:`N_t` is the
+            number of nodes for node type :math:`t`, and :math:`D_t` is the feature
+            size for node type :math:`t`
+        tmp : float
+            The temperature parameter fed to the sampling procedure.
+        kwargs : dict
+            Additional arguments passed to the GNN model.
+
+        Returns
+        -------
+        Tensor
+            A scalar tensor representing the loss.
+        """
+        self.model = self.model.to(graph.device)
+        self.elayers = self.elayers.to(graph.device)
+
+        pred = self.model(graph, feat, embed=False, **kwargs).argmax(-1).data
+
+        prob, hetero_edge_mask = self.explain_graph(
+            graph, feat, tmp=tmp, training=True, **kwargs
+        )
+
+        # store edge mask in homogeneous format
+        self.edge_mask = torch.concat(list(hetero_edge_mask.values()))
```

Comment:
Is this guaranteed to be the order of the node/edge types between the conversion of homogeneous graphs and heterogeneous graphs?

Reply:
Initially I relied on the determinism of the edge dict, but based on the convs. below I think it makes more sense to retain `edge_mask` from the `explain_graph` step and avoid the redundant conversion.

let me know what you think of the changes once they are visible

Reply:
What if we reply on `to_homogeneous` to perform this kind of conversion? In this way, it's guaranteed that the order of edge-specific information is consistent with the result of `to_homogeneous`.

Reply:
will make sure the conversions are always obtained using `to_homogeneous` 👍